docs/topics/api/auth.rst

```diff
@@ -1,7 +1,109 @@
+.. _api-auth:
+
 ================
 Authentication
 ================
 
-To authenticate against this API you will need an API key.
+To access the API, you need to include a `JSON Web Token (JWT)`_
+in an authorization header for every request. This header authenticates
+your user account so you could think of it like a session cookie.
+
+Access Credentials
+==================
+
+To create JWTs, first obtain a **key** and **secret** from the
+`API Credentials Management Page`_.
+Keep the secret string well protected and never check it into version control.
+If someone obtains your secret they can make API requests on behalf of your
+user account.
+
+Create a JWT for each request
+=============================
+
+Prior to making every API request, you need to generate a fresh `JWT`_.
+The JWT will have a short expiration time and is only valid for a single
+request so you can't cache or reuse it.
+You only need to include a few standard fields; here's what the raw JSON object
+needs to look like before it's signed:
+
+.. code-block:: json
+
+    {
+        "iss": "user:543210:23",
+        "iat": 1447273096,
+        "exp": 1447273156,
+    }
+
+iss
+    This is a `standard JWT claim`_ identifying
+    the *issuer*. Set this to the **API key** you generated on the
+    `credentials management page`_.
+    For example: ``user:543210:23``.
+iat
+    This is a `standard JWT claim`_ indicating
+    the *issued at time*. It should be a Unix epoch timestamp and
+    **must be in UTC time**.
+exp
+    This is a `standard JWT claim`_ indicating
+    the *expiration time*. It should be a Unix epoch timestamp in UTC time
+    and must be **no longer than 30 seconds** past the issued at time.
+
+.. note::
+    If you're having trouble authenticating, make sure your system
+    clock is correct and consider synchronizing it with something like
+    `NTP (Network Time Protocol) <http://www.ntp.org/>`_.
+
+Take this JSON object and sign it with the **API secret** you generated on the
+`credentials management page`_. You must sign the JWT using the ``HMAC-SHA256``
+algorithm (which is typically the default).
+The final JWT will be a blob of base64 encoded text, something like::
+
+    eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpYXQiOjE0NDcyNzMwOTZ9.MG9LJiEK5_Db8WpF5cWWRebXCtUB48EJzxKIBqQhSOo
+
+Here is an example of creating a JWT in `NodeJS <https://nodejs.org/en/>`_
+using the `node-jsonwebtoken <https://github.com/auth0/node-jsonwebtoken>`_
+library:
+
+.. code-block:: javascript
+
+    var jwt = require('jsonwebtoken');
+    var secret = 'your-api-secret';  // store this securely.
+
+    var token = jwt.sign({iss: 'user:543210:23'}, secret, {
```

Comment:
Because we're dealing with add-ons I think a NodeJS code example is the most appropriate for the audience. However, I really don't think this library has the nicest interface. It is the leading NodeJS library from what I can tell.